specification/containerservice/resource-manager/Microsoft.ContainerService/preview/2022-06-02-preview/fleets.json

```diff
@@ -0,0 +1,817 @@
+          "description": "The URL to get the next page of Fleet members."
+        }
+      },
+      "description": "The response from the List FleetMembers operation."
+    },
+    "FleetPatch": {
+      "type": "object",
+      "description": "Properties of a Fleet that can be patched.",
+      "properties": {
+        "tags": {
+          "type": "object",
+          "additionalProperties": {
+            "type": "string"
+          },
+          "x-ms-mutability": [
+            "read",
+            "create",
+            "update"
+          ],
+          "description": "Resource tags."
+        }
+      }
+    },
+    "FleetMember": {
+      "type": "object",
+      "description": "A member of the Fleet. It contains a reference to an existing Kubernetes cluster on Azure.",
+      "properties": {
+        "properties": {
+          "description": "Properties of a Fleet member.",
+          "$ref": "#/definitions/FleetMemberProperties",
+          "x-ms-client-flatten": true
+        }
+      },
+      "allOf": [
+        {
+          "$ref": "../../../../../common-types/resource-management/v3/types.json#/definitions/AzureEntityResource"
+        }
+      ]
+    },
+    "FleetMemberProperties": {
+      "description": "Properties of a Fleet member.",
+      "type": "object",
+      "properties": {
+        "clusterResourceId": {
+          "type": "string",
+          "x-ms-mutability": [
+            "read",
+            "create"
+          ],
+          "description": "The ARM resource id of the cluster that joins the Fleet. Must be a valid Azure resource id. e.g.: '/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.ContainerService/managedClusters/{clusterName}'."
+        },
+        "provisioningState": {
+          "readOnly": true,
+          "type": "string",
+          "x-ms-enum": {
+            "name": "FleetMemberProvisioningState",
+            "modelAsString": true
+          },
+          "enum": [
+            "Succeeded",
+            "Failed",
+            "Canceled",
+            "Joining",
+            "Leaving",
+            "Updating"
+          ],
+          "description": "The provisioning state of the last accepted operation."
+        }
+      }
+    },
+    "FleetCredentialResults": {
+      "type": "object",
+      "properties": {
+        "kubeconfigs": {
+          "type": "array",
+          "x-ms-identifiers": [
+            "name"
+          ],
+          "readOnly": true,
+          "items": {
+            "$ref": "#/definitions/FleetCredentialResult"
+          },
+          "description": "Base64-encoded Kubernetes configuration file."
+        }
+      },
+      "description": "The list credential result response."
+    },
+    "FleetCredentialResult": {
+      "type": "object",
+      "properties": {
+        "name": {
+          "type": "string",
+          "readOnly": true,
+          "description": "The name of the credential."
+        },
+        "value": {
```

Comment:
>value [](http://example.com/codeflow?start=9&length=5)

Flag with x-ms-secret extension.

Reply:
We don't do this in AKS for the same API. What are the consequences to the end user?

https://github.com/Azure/azure-rest-api-specs/pull/19653/files/83f13e9bba9320199ccd41bff5fedf8d12b90ef9#diff-3ba6c2915e0cf3cf23e63d80274fa4f20ac009d77fc2623c0d422beebd94fef0R5249

There is nothing secret in here though, because we use AAD to auth the user

Reply:
OK, if the value doesn't hold a secret, the extension isn't needed. The extension is used by ARM Whatif (and possibly other services) to mask secrets. GET responses are not allowed to contain secrets, since secret values must allow for separate RBAC from CRUD operations. If secrets need to be retrieved, it should be via separate POST API. It's OK if the secret property is present in the GET  payload as long as it's blanked/nulled out by the backend. The extension also helps with linting the spec files.

Reply:
the ListCredential is a POST request